### DISCUSSION TITLE
Alert's when a message doesn't arrive in a intermittent window of time.

### DISCUSSION THREAD

**Discussion by michaeljeffreycollins** [nextgenhealthcare/connect#6185] [Q&A]
I know that the subject of alerts in Mirth has been asked quite a lot. I have already looked at 10+ posts in the old forum and the new. I haven't found a solution that even approaches what I need. 

I have a number of processes that run once every 2 weeks, once a month on a given day, etc. The schedule is always a cron job. But when a message doesn't happen, we need some sort of alert. 

Let's take one of the 2 week files as the example.  At some point between 10 AM and 6 PM on that given day one of my end users will drop a file into a network directory and we will process it.  I need to know if no file was ever dropped. We already use the advanced alerting plugin to alert us/end user that there was a problem if the format is wrong as we will explicitly throw an error in the transformer.  In this case I need it on Monday's only. 

What about if a file runs only on the 1st of each month between 2 and 3 AM? The day of the week changes every month. 

The closest solution is based on the No. of messages (sliding window). 

What solutions to this issue are commonly used to resolve it?

**Comment by pacmano1** [comment]
That plugin comes with support from Nexxgen, what did they suggest?

**Comment by rjkroll** [comment]
If you are using a File Reader as a source that will be kinda hard to do, as when it Polls and detects no file, nothing will happen.   You could try using a Javascript reader to monitor the folder and trigger and alert if it detects no files when it runs, or else if there are files, continue with normal processing.   Or, you could use the same File Reader, but at the end, write out the name of the file (if any) processed to a log file, then have a second channel read the log file.  If no records are in the file, no files were sent, send an alert, etc.   You could try using some sort of GlobalVariable that holds the record count for the day it runs, then use another channel that runs at the end of the day and checks if the variable = 0, or >0 etc.
